index.js

```diff
@@ -111,7 +111,9 @@ module.exports = function makeFetch (opts = {}) {
       await archive.ready()
 
       if (version) {
-        archive = archive.checkout(version)
+        const tags = await archive.getAllTags()
+        const tagsObject = Object.fromEntries(tags)
```

Comment:
Would it make sense to use `archive.getTaggedVersion`? https://github.com/hypercore-protocol/hyperdrive#drivegettaggedversionname-cb

Reply:
Also, should we account for versions being actual version numbers in addition to tags? Maybe do a regex to check if it's all numbers and use `archive.checkout(parseInt(version))` instead of doing the tag lookup?

Reply:
Ah! `archive.getTaggedVersion` totally makes sense, I'll check that it works.

Maybe not allow version ids; they're only exposed to users via `fetch(,{method:'TAGS'}` and if a user tried to name a tag a number, they would be very confused as to why their tag isn't working.

Reply:
archive.getTaggedVersion comitted, resolved.

Reply:
Hmmm, I think that versions are pretty important, still.

I think both Gateway and Beaker support them, so it might be a good idea if we did too. 🤔

Maybe we can error out when someone tries to make a tag that's just numbers?